core/vm/contracts.go

```diff
@@ -358,3 +375,49 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	return 1 //TODO: implement
```

Comment:
Even if you don't quite know what constants to use, perhaps you have an idea about the general pricing scheme? Like, is it `rounds * K *input_size` ? Would be good to have _something_, even if we determine the constants later. It would make it easier for others people to help tune the constants.

Reply:
The benchmarks so far lead me to believe that a round is ~ 1 gas (!), and that this precompile will be dominated by calling cost. We're working on a more robust set of benchmarks that will include input size as a floating param to inform gas.

More discussion [here](https://github.com/keep-network/EIPs/pull/2/files#discussion_r295576690)

Reply:
I believe the methodology of pricing relative to `ecrecover` is sound. Once we have a wider range of inputs on benchmarks this will be straightforward.

Reply:
I opened this PR in an early-mode before we had a chance to discuss benchmarks and gas price. Based on what we know so far, and using the same methodology as for the [EIP-1108](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1108.md) it seems that it should be [1 gas/round](https://github.com/ethereum/EIPs/pull/2129#discussion_r296223207).

I was hoping to remove this TODO and put a final value here after we agree on the gas [in the EIP](https://github.com/ethereum/EIPs/pull/2129). I am not sure how the process should look here.

Reply:
I set the gas price to 1/round based on our benchmarks.